### PR TITLE
chore: run e2e tests in github actions

### LINKED ISSUE #6: Running existing tests inside GitHub Actions
This repository has extensive E2E tests, however we don't run them inside GitHub Actions.

We should setup a proper CI pipeline that building the local Heighliner image, and runs those tests.

### PR DESCRIPTION
**Pull request by boojamya**
closes #6 

## Summary by CodeRabbit

- **New Features**
	- Introduced a Docker ignore configuration to enhance Docker build efficiency.
	- Added new GitHub Actions workflows for automated end-to-end and unit testing.
	- Updated configuration to specify build directories in `chains.yaml`.

- **Bug Fixes**
	- Removed an unnecessary version specification in end-to-end test configuration to streamline testing procedures.

- **Chores**
	- Adjusted Makefile to refine build target settings for better build process management.

### CODE REVIEW

File: simapp/Makefile [hunk 1]
```diff
@@ -24,4 +24,4 @@ BUILD_FLAGS := -ldflags '$(ldflags)'
 build:
 	@go build -mod=readonly $(BUILD_FLAGS) -o $(PWD)/build/ ./...
 
-.PHONY: build
+.PHONY: build
```
Comment (johnletey) [inline simapp/Makefile]: Let's add back the newline here to minimize changes 😄

```suggestion
.PHONY: build

```

File: chains.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 - name: noble-forwarding-simd
   dockerfile: cosmos
   build-target: make build
+  build-dir: simapp
```
Comment (johnletey) [inline chains.yaml:4]: @boojamya Is this a required option in a new version of Heighliner?

**Review by johnletey — COMMENTED** [review]
@boojamya Let's chat in our sync today about this. I was under the impression that we could simply use the [`heighliner-build-action`](https://github.com/strangelove-ventures/heighliner-build-action), but also like the approach of a simple Dockerfile.

Either way, could we get newlines added to all end of files? See one of my comments below.

File: chains.yaml (hunk 2 above)
Comment (boojamya) [reply chains.yaml:4]: No, it's not. I just thought this was more straight forward. 

Right now, `make local-image` runs heighliner build, which checks the `chains.yaml` file, which runs `make build` (using root Makerfile), which cd's into simapp and then runs the simapp `make build`.

Adding `build-dir` will streamline this and heighliner will just run `make build` from the simapp folder. 

Happy to revert if you would like to keep original flow.

File: .github/workflows/e2e-tests.bak [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        uses: actions/setup-go@v4
+        with:
+          go-version: '^1.22'
+
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+          # Run the command and convert its output to a JSON array
+          TESTS=$(cd e2e && go test -list . | grep -v "^ok " | jq -R -s -c 'split("\n")[:-1]')
+          echo "matrix=${TESTS}" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (boojamya) [reply .github/workflows/e2e-tests.bak]: All of the github actions documentation I've seen notes using the $GITHUB_OUTPUT variable.

Ref: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

**Comment by boojamya** [comment]
> @boojamya Let's chat in our sync today about this. I was under the impression that we could simply use the [`heighliner-build-action`](https://github.com/strangelove-ventures/heighliner-build-action), but also like the approach of a simple Dockerfile.
> 
> Either way, could we get newlines added to all end of files? See one of my comments below.

After a lot of testing, the action is working!

It required the SL team cutting a new release of heighliner as the old release was not compatible with go 1.22. For context, the action automatically uses the latest release of heighliner.

**Review by johnletey — APPROVED** [review]

**Pull request merged**